uni/lib/view/login/login.dart

```diff
@@ -94,22 +98,28 @@ class LoginPageViewState extends State<LoginPageView> {
     }
   }
 
-  /// Tracks if the user wants to keep signed in (has a
-  /// checkmark on the button).
-  void _setKeepSignedIn({bool? value}) {
-    if (value == null) {
-      return;
-    }
-    setState(() {
-      _keepSignedIn = value;
-    });
-  }
+  Future<void> _falogin(BuildContext context) async {
+    final stateProviders = StateProviders.fromContext(context);
+    final sessionProvider = stateProviders.sessionProvider;
 
-  /// Makes the password input view hidden.
-  void _toggleObscurePasswordInput() {
-    setState(() {
-      _obscurePasswordInput = !_obscurePasswordInput;
-    });
+    try {
+      setState(() {
+        _loggingIn = true;
+      });
+      await sessionProvider.federatedAuthentication(
+        context,
+        persistentSession: _keepSignedIn,
+      );
+    } catch (err, _) {
+      await closeInAppWebView();
+      setState(() {
+        _loggingIn = false;
+      });
+      Logger().e('Failed to authenticate');
+      if (context.mounted) {
+        unawaited(ToastMessage.error(context, 'Failed to authenticate'));
+      }
```

Comment:
If we want as many people as possible to use federated authentication, we should collect any exceptions that occur in this flow and send them to sentry, so we can fix them.

Reply:
So, throw the error again?

Reply:
Not necessarily, you can use the Sentry.captureExpection method (or something similiar) to send the exception to Sentry